gax/src/main/java/com/google/api/gax/rpc/ServerStreamingCallSettings.java

```diff
@@ -52,11 +113,120 @@ private ServerStreamingCallSettings(Builder<RequestT, ResponseT> builder) {}
 
   public static class Builder<RequestT, ResponseT>
       extends StreamingCallSettings.Builder<RequestT, ResponseT> {
+    private Set<StatusCode.Code> retryableCodes;
+    private RetrySettings retrySettings;
+    private StreamResumptionStrategy<RequestT, ResponseT> resumptionStrategy;
 
-    private Builder() {}
+    private Duration timeoutCheckInterval;
+    private Duration idleTimeout;
+
+    /** Initialize the builder with default settings */
+    private Builder() {
+      this.retryableCodes = ImmutableSet.of();
+      this.retrySettings = RetrySettings.newBuilder().build();
+      this.resumptionStrategy = new SimpleStreamResumptionStrategy<>();
+
+      this.timeoutCheckInterval = Duration.ZERO;
```

Comment:
It seems like `null` is allowed in setter methods (for this and other fields of this builder class, see `setTimeoutCheckInterval()` for example). In such case it seems leaving those field as `null` in case if default constructor called should be ok (and better). Another option would be to add validation on setters or (I strongly believe as the most preferable) add validation in (and ONLY in) constructed class  (`ServerStreamingCallSettings`). This would ensure that an invalid object cannot be constructed (regardless of how the constructor is called: via builder, directly, via reflection), and also will allow "invalid" intermediate states of a builder, which should be ok, because builders may collect data over time and do not have all needed information on the moment of creation of the builder itself (thus should allow "invalid" state until build() is called).

Reply:
I added precondition checks on the setters. I think it's a better user experience to get the call stack of how the null entered the system